main.cpp

```diff
@@ -374,6 +375,27 @@ void receivePing (udp_client_server::udp_server& server)
     std::cerr << "Received start signal: " << buff << "\n";
 }
 
+//r = distance, theta = heading and phi = angular displacement relative to the target
+void sendData(y2017::vision::vision_message *vison_writer, struct& message) {
```

Comment:
Also, y2017 is not connected to the vision code. On the vision side, you should only be writing to a protobuf object, serializing it, and then sending it over UDP (possibly copy pasting 971's UDP code in aos).